air/src/stack/mod.rs

```diff
@@ -23,6 +24,20 @@ const B1_COL_IDX: usize = B0_COL_IDX + 1;
 /// inputs and outputs as well as the initial values of the bookkeeping columns.
 pub const NUM_ASSERTIONS: usize = 2 * MIN_STACK_DEPTH + 2;
 
+/// The number of generic constraints in the stack operations.
+pub const NUM_GENERIC_CONSTRAINTS: usize = 16;
+
+/// The degrees of constraints in the generic stack operations.
+pub const CONSTRAINT_DEGREES: [usize; NUM_GENERIC_CONSTRAINTS] = [
+    // Each of the operation are being multiplied with a flag which can be at max of 7 degree.
+    // There are 3 ways in which an operation can effect the stack. It can shift each stack item
+    // to the right or left or doesn't shift it at all. Given only one of the three state transition
+    // is possible, these 3 transition of an item can be combined into one constraint. Therefore, there
+    // will be 16 generic stack constraints encompassing all the 16 stack items. 7 is to each one of
+    // them.
```

Comment:
nit: I'd probably move this into the doc comment above the constant declaration. Also, last sentence is a bit confusing ("7 is to each...") or at least I didn't understand what it means.